### PR TITLE
Spacing presets: Modify the styling of the input controls when in unlinked mode in order to better differentiate sides

### LINKED ISSUE #44121: Update spacing in exploded padding control
Part of https://github.com/WordPress/gutenberg/issues/43412

The current spacing is poor because the range is equidistant from the label before and after, which can make it hard to understand which control and which label are related:

Screenshot 2022-09-13 at 13 06 02

It would be good to update the spacing so that there's a larger gap between the control and the subsequent label. 

Screenshot 2022-09-13 at 13 06 39

Considering https://github.com/WordPress/gutenberg/issues/41973 it would also be good to increase the height of the input itself to 40px instead of the current 30px.

### PR DESCRIPTION
**Pull request by glendaviesnz**
## What?
Modifies CSS on spacing presets control to better differentiate which side a control relates to. Also adds a slight indent to better indicate which dimension setting they relate to

## Why?
Currently the control is equidistant from the above and below labels so hard to tell which one it relates to.
Fixes: #44121

## How?
Adds some additional margins and padding

## Testing Instructions
 - Add a Group and set unlinked Padding and Margins and check that controls display as expected for both preset and custom values

## Screenshots or screencast 

Before:

before

After:

Screen Shot 2022-09-29 at 4 01 15 PM

### CODE REVIEW

**Comment by jasmussen** [comment]
Nice, a big improvement, thanks for working on this:

Screenshot 2022-09-14 at 10 27 36

I don't think we should do the indent, though, as it breaks [the vertical](https://github.com/WordPress/gutenberg/issues/43423#issuecomment-1232540038) grid we're aiming for. Can we update the labels to be the new all-caps style as well? I tried myself replacing the `Text` component used for labels at the moment, with the `StyledLabel` version instead, but I ran into a wall with the component import. Possible?

**Comment by glendaviesnz** [comment]
@jasmussen thanks for reviewing, will look at implementing the suggested changes tomorrow - can you point me to a control that has the all caps style already?

**Comment by glendaviesnz** [comment]
@jasmussen I have made the suggested changes. 

Screen Shot 2022-09-15 at 11 41 29 AM

I just copied the font control and used `BaseControl.VisualLabel` to get the uppercase as it looks like `StyledLabel` is not exported from components yet.

**Comment by jasmussen** [comment]
Nice, thank you. I'd love a quick review by @jameskoster as he has the metrics on mind.

The icon looks bottom-aligned compared to the label, can it be centered vertically with it?

**Comment by jameskoster** [comment]
Better, thank you! But there are a few bolts to tighten.

---

Screenshot 2022-09-15 at 10 39 53

The range doesn't seem to occupy the full container width – there's a gap between the grey track and the right-hand edge.

---

Screenshot 2022-09-15 at 10 42 55


Range height is 36px, can we make it 40px, with a 16px bottom margin?

---

Screenshot 2022-09-15 at 10 44 17


When you switch to a custom value, a gap appears on the left/right.

--- 

The label container has a 30px min-height. It should instead wrap the label (and consequently have an inherent height of 16px), then have a bottom margin of 8px to create the correct space between label + range.

Screenshot 2022-09-15 at 10 46 36

---

Screenshot 2022-09-15 at 10 49 10

The 'custom' icon for block spacing isn't aligned with the others.

**Comment by glendaviesnz** [comment]
@jameskoster I think I have covered all those things off. There seems like quite a bit of white space now, but maybe that is how you want, or maybe I have missed something:

Screen Shot 2022-09-16 at 4 14 32 PM

**Comment by jameskoster** [comment]
Thanks Glen.

The label is still a bit too tall:

Screenshot 2022-09-20 at 10 19 44

The spacing is based on a scheme @jasmussen has shared before:

Screenshot 2022-09-20 at 10 17 04

It looks a bit large here because the range visually occupies a relatively small vertical space. I don't think it's a deal-breaker, but I'll let Joen decide :)

**Comment by glendaviesnz** [comment]
I think it is a bit closer:

Screen Shot 2022-09-23 at 2 42 38 PM

I had to do it without adding a bottom margin to the label though as this threw out the vertical alignment with the custom value icon on the right.

**Comment by jameskoster** [comment]
@jasmussen How are you feeling about this? Visually it's better, but the spacing doesn't align to the grid correctly. I suspect it's tricky to get right due to the `grid` layout. 

Screenshot 2022-09-23 at 10 04 08

@glendaviesnz I don't know if it's helpful but [here's a Figma](https://www.figma.com/file/9wDJslLfdeDHxC1ula4lgG/Preset-spacing?node-id=1%3A226) with everything spaced according to the grid.

**Comment by glendaviesnz** [comment]
Thanks @jameskoster, yeh the grid layout makes it a bit trickier, but I should be able to get it much closer with this extra detail you have provided - public holiday here Monday though, but should be able to sort first thing Tuesday

**Comment by glendaviesnz** [comment]
@jameskoster I had to use some margins to get the spacing rather than the grid gaps as it isn't possible to specify different gaps on a per row basis (as far as I am aware). It it pretty close to the Figma design now - the following is the current layout on this PR pasted over the figma one:

figma

The range is about 0.5 pixels out, but this is the closest I could get it.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss]: `$grid-unit-15`.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss]: `$grid-unit-20`.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss]: `$grid-unit-15`.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss]: `$grid-unit-15`.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss]: `$grid-unit-20`.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss:104]: Should we use `$grid-unit-50` here? Unsure how the other control updates have specified their height.

**Review by jameskoster — COMMENTED** [review]
Thanks for your perseverance with this one Glen, it's nicely aligned to the grid now :) 

I left a few comments, perhaps we can use variables for some of the values?

The last thing we need to decide on from the design perspective is whether to pursue the 40px control height for the ranges here. It does create a large gap. But if we reduce the height it will likely require tweaking other spacing values. What do you think @jasmussen ?

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (glendaviesnz) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:104]: At the moment [color indicator](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/color-indicator/style.scss#L2) is the only component using that var, and it does a `*0.5` on it to get `20px`.

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (glendaviesnz) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss]: yeh, probably makes sense to use the vars for these

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (jameskoster) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:104]: Perhaps it doesn't matter too much at this point. The Spacing Range component may need to be refactored a bit to align with other controls? cc @ciampo @mirka

Comment (ciampo) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:104]: Let's _not_ use `$grid-unit-50` — it's not something that we've been using much in `@wordpress/components` (color indicator is a legacy component and in this case a bit of an exception)

**Review by ciampo — COMMENTED** [review]
Had a quick look at this PR and this file, noticed a few things:

- We should use the `__nextHasNoMarginBottom` prop on `RangeControl` — this way the `RangeControl` component doesn't add any bottom margin by default, and it makes it easier to add spacing around it
- The labelling situation of composite range controls + unit controls looks a bit messy, we should probably start allocating resources for working on https://github.com/WordPress/gutenberg/issues/42630#issuecomment-1233937028
- It looks like we're applying a few style overrides to `RangeControl` (most importantly, setting a `40px` height) — these changes/necessities ideally should be discussed in a separate issue and applied directly to `RangeControl` instead of as a style override (cc'ing @mirka who has been working on updating control component sizes in the past months)

**Comment by glendaviesnz** [comment]
> We should use the __nextHasNoMarginBottom prop on RangeControl

I have done this and adjusted things to maintain the same layout as before. The Range control adjustments were only ever temporary while @aaronrobertshaw was working on the new version of the control. I think we should merge this in as is if we want to get it into 6.1 and follow up on other things in separate issues as suggested

**Review by ramonjd — COMMENTED** [review]
Here's what I see in the post/site editors:

Screen Shot 2022-09-29 at 12 22 48 pm

The spacing between the label and above + below unlinked controls seems to be the same and not like the "After" image in the description.

Screen Shot 2022-09-29 at 12 31 35 pm

**Comment by ramonjd** [comment]
> I have done this and adjusted things to maintain the same layout as before.

Or does this mean that my above comment isn't valid?

**Comment by glendaviesnz** [comment]
Sorry @ramon the before and after had not kept up with the ongoing changes - have updated to 

Screen Shot 2022-09-29 at 4 01 15 PM

which matches the Figma layout requested:

Screen Shot 2022-09-29 at 4 03 14 PM

**Review by ramonjd — APPROVED** [review]
LGTM
Thanks for the update @glendaviesnz !

**Comment by ciampo** [comment]
> The Range control adjustments were only ever temporary while @aaronrobertshaw was working on the new version of the control. I think we should merge this in as is if we want to get it into 6.1 and follow up on other things in separate issues as suggested

Sure, my comments weren't mean to be a blocker — but still, it would be good to discuss any tweaks to `RangeControl`'s design (like a new `40px` height and any other vertical spacing adjustment) in a separate issue and apply those changes directly to the component (`RangeControl` and/or the new `SliderControl` that aaron has been working on)

Same goes for

> The labelling situation of composite range controls + unit controls looks a bit messy, we should probably start allocating resources for working on https://github.com/WordPress/gutenberg/issues/42630#issuecomment-1233937028

As we build more complex composite controls, we should really have a solid way of labelling such controls consistently across the editor

**Comment by michalczaplinski** [comment]
Cherry-picked for the 6.1 release in https://github.com/WordPress/gutenberg/pull/44656

**Pull request merged**
